src/test/java/edu/wisc/my/messages/data/MessagesFromTextFileTest.java

```diff
@@ -4,13 +4,44 @@
 import static org.mockito.Mockito.when;
 
 import java.io.IOException;
+
+import org.junit.Assert;
 import org.junit.Test;
+
+import org.springframework.mock.env.MockEnvironment;
+
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.core.env.Environment;
+import org.springframework.core.io.DefaultResourceLoader;
 import org.springframework.core.io.Resource;
 import org.springframework.core.io.ResourceLoader;
 
 public class MessagesFromTextFileTest {
+  
+  /**
+   * Test that the data in the incoming text file is actually valid.
+   */
+  @Test
+  public void dataIsValid() {
+
+    MockEnvironment mockEnv = new MockEnvironment();
+    
+    // ToDo: It would be very cool is this value could automaticall stay in sync
+    //  with the actual value in the application.properties file.
+    mockEnv.setProperty("message.source", "classpath:messages.json");
+    
+    ResourceLoader defaultLoader = new DefaultResourceLoader();
 
+    MessagesFromTextFile messageReader = new MessagesFromTextFile();
+    messageReader.setEnv(mockEnv);
+    messageReader.setResourceLoader(defaultLoader);
+    try{
+       messageReader.allMessages();
+    } catch (Exception e) {
```

Comment:
Suggestion: don't `catch` here. Let the Exception throw from `dataIsValid()`. The unhandled exception will fail the test, and may be useful in the detail it exposes about why `messageReader.allMessages()` failed.